File: lab2/py/FizzBuzzCounter.py
number = int(input("Please enter a number between 10 and 100: "))
while number < 10 or number > 100:
    number = int(input("Invalid input. Please enter a number between 10 and 100: "))

fizzCount = 0
buzzCount = 0
fizzBuzzCount = 0

for i in range(1, number + 1):
    if i % 7 == 0:
        print(f"({i} is skipped)")
        continue
    elif i % 5 == 0 and i % 3 == 0:
        print("FizzBuzz")
        fizzBuzzCount += 1
    elif i % 5 == 0:
        print("Buzz")
        buzzCount += 1
    elif i % 3 == 0:
        print("Fizz")
        fizzCount += 1
    else:
        print(i)

print("\n--- Summary ---")
print(f"Fizz count: {fizzCount}")
print(f"Buzz count: {buzzCount}")
print(f"FizzBuzz count: {fizzBuzzCount}")

File: lab2/cpp/FizzBuzzCounter.cpp
#include <iostream>

using namespace std;

int main() {
  int n = 0; // Initialize n to 0 for the first prompt condition

  while (true) {
    if (n == 0) { // İlk mesaj
      cout << "Please enter a number between 10 and 100: ";
    } else {
      cout << "Invalid input. Please enter a number between 10 and 100: ";
    }

    cin >> n;

    if (n >= 10 && n <= 100) {
      break;
    }
  }

  int fizzCount = 0;
  int buzzCount = 0;
  int fizzBuzzCount = 0;

  for (int i = 1; i <= n; i++) {
    if (i % 7 == 0) {
      cout << "(" << i << " is skipped)\n";
      continue;
    } else if (i % 3 == 0 && i % 5 == 0) {
      cout << "FizzBuzz\n";
      fizzBuzzCount++;
    } else if (i % 3 == 0) {
      cout << "Fizz\n";
      fizzCount++;
    } else if (i % 5 == 0) {
      cout << "Buzz\n";
      buzzCount++;
    } else {
      cout << i << "\n";
    }
  }

  cout << "--- Summary ---\n";
  cout << "Fizz count : " << fizzCount << "\n";
  cout << "Buzz count : " << buzzCount << "\n";
  cout << "FizzBuzz count: " << fizzBuzzCount << "\n";

  return 0;
}